src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,231 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+
+    private final DriverDao driverDao = new DriverDaoImpl();
+    private final ManufacturerDao manufacturerDao = new ManufacturerDaoImpl();
+
+    @Override
+    public Car create(Car car) {
+        String createRequest = "INSERT INTO cars (model, manufacturer_id) VALUES (?,?)";
+        manufacturerDao.create(car.getManufacturer());
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                        connection.prepareStatement(createRequest,
+                             PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                Long id = resultSet.getObject(1, Long.class);
+                car.setId(id);
```

Comment:
```suggestion
            if (resultSet.next()) {
                car.setId(resultSet.getObject(1, Long.class));
```

your code is working but by using while you may confuse others that there'll be more than one result